From 894b1179de918c98c8eea51a3b166a200b0bcea2 Mon Sep 17 00:00:00 2001
From: Yelena Krivosheev <yelena@marvell.com>
Date: Mon, 30 Jul 2018 15:21:57 +0300
Subject: [PATCH 0676/1051] mvneta: improve rx swbm performance by copy less

Copy only first 'rx_header_size' bytes insted
'rx_copybreak' bytes for packets >= 'rx_copybreak'.

Change-Id: I635c9561f181cb00f81e6a299d894fc69787eb29
Signed-off-by: Yelena Krivosheev <yelena@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvneta.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvneta.c b/drivers/net/ethernet/marvell/mvneta.c
index a6dba2cbed94..a7eb0ff35e40 100644
--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -1981,7 +1981,10 @@ static int mvneta_rx_swbm(struct napi_struct *napi,
 				rxq->skb_alloc_err++;
 				continue;
 			}
-			copy_size = min(skb_size, rx_bytes);
+			if (skb_size > rx_bytes)
+				copy_size = rx_bytes;
+			else
+				copy_size = rx_header_size;
 
 			/* Copy data from buffer to SKB, skip Marvell header */
 			memcpy(rxq->skb->data, data + MVNETA_MH_SIZE,
-- 
2.17.1

